chain/app/db_app.go

```diff
@@ -39,6 +50,18 @@ func (app *dbBandApp) DeliverTx(req abci.RequestDeliverTx) (res abci.ResponseDel
 	return response
 }
 
-func (app *dbBandApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeginBlock) {
+func (app *dbBandApp) BeginBlock(req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+	// Open transaction
+	// if err := app.dbBand.ValidateChainID(app.Ba)
+	app.dbBand.OpenTransaction()
+
 	return app.bandApp.BeginBlock(req)
 }
+
+func (app *dbBandApp) EndBlock(req abci.RequestEndBlock) abci.ResponseEndBlock {
+	res := app.bandApp.EndBlock(req)
```

Comment:
We need to call `app.bandApp.EndBlock` after we are done with logic. That's where the context is reset.